cmd/vic-machine/create/create.go

```diff
@@ -874,16 +877,41 @@ func (c *Create) processDNSServers() error {
 
 func (c *Create) processVolumeStores() error {
 	defer trace.End(trace.Begin(""))
-	c.VolumeLocations = make(map[string]string)
+	c.VolumeLocations = make(map[string]*url.URL)
 	for _, arg := range c.volumeStores {
-		if err := common.CheckUnsupportedCharsDatastore(arg); err != nil {
-			return fmt.Errorf("--volume-store contains unsupported characters: %s Allowed characters are alphanumeric, space and symbols - _ ( ) / :", err)
+		splitMeta := strings.Split(arg, ":")
+		if len(splitMeta) < 1 {
+			return errors.New("volume store input must be in format datastore/path:label or <nfs://<host>/<url-path>?<mount option as query parameters>:,<comma separated mount options>label")
+		}
+
+		// divide out the label with the target
+		lastIndex := len(splitMeta)
+		label := splitMeta[lastIndex-1]
+		rawTarget := strings.Join(splitMeta[0:lastIndex-1], ":")
+
+		// raw target input should be in the form of a url
+		urlTarget, err := url.Parse(rawTarget)
+		if err != nil {
+			return fmt.Errorf("parsed url for option --volume-store does not have scheme 'nfs',  valid inputs are datastore/path:label or <nfs://<host>/<share point path>?<mount options as query params>:label")
```

Comment:
looks like the valid format error message is used multi places, and you had one with a bit problem (no colon before label^^^), so suggest to define a variable for the common error message